cmd/gen-crl/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/pkcs11helpers"
+)
+
+type crlDef struct {
+	num   int
+	start time.Time
+	end   time.Time
+}
+
+func generateDefs(clk clock.Clock, total, initialNum int, startStr string, period, overlap time.Duration) ([]crlDef, error) {
```

Comment:
Please add a function comment describing what this does, and each of the parameters. Also, let's use a longer name (short names are best only when the scope is narrow).